lib/fluent/plugin/parser_syslog.rb

```diff
@@ -27,8 +27,9 @@ class SyslogParser < Parser
       REGEXP = /^(?<time>[^ ]*\s*[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[^ :\[]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$/
       # From in_syslog default pattern
       REGEXP_WITH_PRI = /^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[^ :\[]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$/
-      REGEXP_RFC5424 = /\A(?<time>[^ ]+) (?<host>[!-~]{1,255}) (?<ident>[!-~]{1,48}) (?<pid>[!-~]{1,128}) (?<msgid>[!-~]{1,32}) (?<extradata>(?:\-|\[(.*)\]))(?: (?<message>.+))?\z/m
-      REGEXP_RFC5424_WITH_PRI = /\A\<(?<pri>[0-9]{1,3})\>[1-9]\d{0,2} (?<time>[^ ]+) (?<host>[!-~]{1,255}) (?<ident>[!-~]{1,48}) (?<pid>[!-~]{1,128}) (?<msgid>[!-~]{1,32}) (?<extradata>(?:\-|\[(.*)\]))(?: (?<message>.+))?\z/m
+      REGEXP_RFC5424 = "(?<time>[^ ]+) (?<host>[!-~]{1,255}) (?<ident>[!-~]{1,48}) (?<pid>[!-~]{1,128}) (?<msgid>[!-~]{1,32}) (?<extradata>(?:\\-|(?:\\[.*?(?<!\\\\)\\])+))(?: (?<message>.+))?"
+      REGEXP_RFC5424_NO_PRI = Regexp.new('\\A' + REGEXP_RFC5424 + '\\z', Regexp::MULTILINE)
```

Comment:
`\` for escaping seems not to be necessary since `p Regexp.new('\\A', Regexp::MULTILINE) == Regexp.new('\A', Regexp::MULTILINE)` reutrns `true`

Reply:
They work the same. I preferring always escape it for readability, so people who come from other languages (like me) can distinguish string literal from control chars easily. What do you think of it?

Reply:
It's IMO but I prefer the one without escape because the current value is so.
I can't think that `(?:\\-|(?:\\[.*?(?<!\\\\)\\])+))` is readeable...

Reply:
Agreed. I tried to make it more readable by using Heredoc :)

Reply:
Is there any reason to change `(?<extradata>(?:\-|(?:\[.*?(?<!\\)\])+))` ?
it makes sense to me for adding `+`  after `\])`. but I can't see the point of adding the part of `?(?<!\\)`.

Reply:
`+` applies for the whole group, because the message can carry mutiple `SD-ELEMENT` .

`(?<!...)` is a negative lookbehind to not match `\]` as an end of SD-ELEMENT since `]` is allowed in SD PARAM-VALUE by escaping with a `\` prefix